app/queries/thread_search_query.rb

```diff
@@ -1,147 +0,0 @@
-    discussion_id = result['id']
-    SearchResult.new model_cache.discussion_for(discussion_id),
-                     model_cache.motion_for(discussion_id),
-                     model_cache.comment_for(discussion_id),
-                     model_cache.discussion_blurb_for(discussion_id),
-                     model_cache.motion_blurb_for(discussion_id),
-                     model_cache.comment_blurb_for(discussion_id),
-                     result['query'],
-                     result['rank']
-  end
-
-  def top_results
-    @top_results ||= SearchVector.execute_search_query self.class.search_sql(visible_results), query: @query, offset: @offset, limit: @limit
-  end
-
-  def visible_results
-    @visible_results ||= Queries::VisibleDiscussions.new(user: @user, groups: @user.groups).pluck(:id)
-  end
-
-  def relevant_records_for(model)
-    return [] unless Rails.application.secrets.advanced_search_enabled
-    SearchVector.execute_search_query self.class.send(:"relevant_#{model}_sql", top_results.map { |d| d['id'] }), query: @query
-  end
-
-  def self.index_thread_sql
-    "INSERT INTO discussion_search_vectors (discussion_id, search_vector)
-     SELECT      id, #{field_weights_sql(discussion_field_weights)}
-     FROM        discussions
-     LEFT JOIN (
-       SELECT string_agg(name, ',')                     AS motion_names,
-              LEFT(string_agg(description, ','), 10000) AS motion_descriptions
-       FROM   motions
-       WHERE  discussion_id = :id) motions ON :id = id
-     LEFT JOIN (
-       SELECT LEFT(string_agg(body, ','), 200000)       AS comment_bodies
-       FROM   comments
-       WHERE  discussion_id = :id) comments ON :id = id
-     WHERE    id = :id"
-  end
-
-  def self.search_sql(visible_ids)
-    "SELECT id,
-            rank,
-            :query as query,
-            #{field_as_blurb_sql('discussions.description')} as blurb
-     FROM (
-        SELECT   discussion_id, search_vector, #{rank_sql} as rank
-        FROM     discussion_search_vectors
-        WHERE    search_vector @@ plainto_tsquery(:query)
-        AND      discussion_id IN (#{visible_ids.join(',')})
-        ORDER BY rank DESC
-        OFFSET   :offset
-        LIMIT    :limit
-     ) vectors
-     INNER JOIN discussions ON discussions.id = vectors.discussion_id
-     WHERE      rank > 0
-     ORDER BY   rank DESC, last_activity_at DESC"
-  end
-
-  def self.relevant_motions_sql(top_ids)
-    "SELECT   DISTINCT ON (discussion_id)
-              id,
-              discussion_id,
-              name,
-              #{rank_sql(motion_field_weights)} as rank,
-              #{field_as_blurb_sql('description')} as blurb
-     FROM     motions
-     WHERE    motions.discussion_id IN (#{top_ids.join(',')})
-     AND      #{field_weights_sql(motion_field_weights)} @@ plainto_tsquery(:query)
-     ORDER BY discussion_id, rank DESC"
-  end
-
-  def self.relevant_comments_sql(top_ids)
-    "SELECT   DISTINCT ON (discussion_id)
```

Comment:
I got rid of the relevant comments / relevant motions stuff for now. It can go back in if it needs to, but it doesn't need to at the moment; I'd rather opt for getting it blazing fast and then make speed sacrifices for quality then if we need to.